pkg/controllers/subnetset/subnetset_controller.go

```diff
@@ -288,33 +295,75 @@ func (r *SubnetSetReconciler) CollectGarbage(ctx context.Context) {
 	}
 }
 
-func (r *SubnetSetReconciler) DeleteSubnetForSubnetSet(obj v1alpha1.SubnetSet, updataStatus bool) (bool, error) {
+func (r *SubnetSetReconciler) deleteSubnetBySubnetSetName(subnetSetName, ns string) error {
+	nsxSubnets := r.SubnetService.ListSubnetBySubnetSetName(ns, subnetSetName)
+	return r.deleteStaleSubnets(nsxSubnets)
+}
+
+func (r *SubnetSetReconciler) deleteSubnetForSubnetSet(obj v1alpha1.SubnetSet, updateStatus bool) error {
 	nsxSubnets := r.SubnetService.SubnetStore.GetByIndex(servicecommon.TagScopeSubnetSetCRUID, string(obj.GetUID()))
-	hitError := false
-	hasStaleSubnetPorts := false
-	for _, subnet := range nsxSubnets {
-		r.SubnetService.LockSubnet(subnet.Path)
-		portNums := len(r.SubnetPortService.GetPortsOfSubnet(*subnet.Id))
+	if err := r.deleteStaleSubnets(nsxSubnets); err != nil {
```

Comment:
Use deleteSubnets as the nsxSubnets is fetch with subnet set UID instead of name so there is no new subnets are included?

Reply:
updated.